docker/README.md

```diff
@@ -0,0 +1,54 @@
+# Neurodamus Docker Image
+
+## Requirements
+You must have:
+* [Docker Desktop](https://www.docker.com/) installed and running.
+* [Git](https://git-scm.com/)
+
+## Docker image
+You can either build your own docker image using the recipe in this repository or pull the prebuilt image from the Docker Hub.
+
+### Option 1: Build your docker image
+Before building your own docker image, make sure that your Docker Desktop is configured with at least 4 GB memory and 2 GB swap. (Settings->Resources->Memory)
+```
+cd docker/
+docker build -t neurodamus .
+```
+### Option 2: Pull the prebuilt image from Docker Hub
+```
+docker pull bluebrain/neurodamus
+```
+You can verify your image with the command `docker image ls`, for example:
+```
+$ docker image ls
+REPOSITORY                            TAG       IMAGE ID       CREATED        SIZE
+bluebrain/neurodamus                  latest    4784d73155e7   11 hours ago   4.08GB
+```
+## Docker containier
+With the docker image, you can start a neurodamus container with an interative Bash shell and meanwhile mount your local folder which contains your mod files and the circuit data.
+```
+docker run --rm -it -v <folder_mods_circuit>:/mnt/mydata bluebrain/neurodamus
+```
+In the Bash shell, first build your mods:
+```
+cd /mnt/mydata/
+cp $EXTRA_MODS_DIR/* <your_mod_dir>/
+build_neurodamus.sh <your_mod_dir>/
+source $USR_VENV/bin/activate
+```
+
+If you have additional hoc files, copy them to the folder `$HOC_LIBRARY_PATH`
+```
+cp /mnt/mydata/<your_hoc_dir>/* $HOC_LIBRARY_PATH/
+```
+
+Then you are ready to start the simulation.
+
+To run with a single process
+```
+./x86_64/special -mpi -python $NEURODAMUS_PYTHON/init.py --configFile=<your_folder>/simulation_config.json
```

Comment:
nitpick: just to avoid confusion, inside docker container, is `-mpi` required if we are not running with srun/mpirun? (HPE-MPI had specific issue)

Reply:
indeed `-mpi` is not required for single process. I've removed it.

Reply:
@WeinaJi : just mentioning the error that Kerem got without -mpi:

> [STEP] Reports Enabling
> [INFO]  * soma (Type: compartment, Target: Mosaic)
> Attempting to use an MPI routine (internal_Comm_rank) before initializing or after finalizing MPICH

So -mpi is required when reports are involved?